src/sentry/sentry_apps/api/endpoints/sentry_app_webhook_requests.py

```diff
@@ -0,0 +1,153 @@
+        default=datetime.strptime("2000-01-01 00:00:00", date_format).replace(tzinfo=timezone.utc),
+        default_timezone=timezone.utc,
+        required=False,
+    )
+    end = serializers.DateTimeField(
+        format=date_format, default=None, default_timezone=timezone.utc, required=False
+    )
+
+    def validate(self, data):
+        if "start" in data and "end" in data and data["start"] > data["end"]:
+            raise serializers.ValidationError("Invalid timestamp (start must be before end).")
+        return data
+
+    def validate_end(self, end):
+        if end is None:
+            end = datetime.now(tz=timezone.utc)
+        return end
+
+
+@control_silo_endpoint
+class SentryAppWebhookRequestsEndpoint(SentryAppBaseEndpoint):
+    owner = ApiOwner.INTEGRATIONS
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (SentryAppStatsPermission,)
+
+    def get(self, request: Request, sentry_app: SentryApp) -> Response:
+        """
+        :qparam string eventType: Optionally specify a specific event type to filter requests
+        :qparam bool errorsOnly: If this is true, only return error/warning requests (300-599)
+        :qparam string organizationSlug: Optionally specify an org slug to filter requests
+        :qparam string start: Optionally specify a date to begin at. Format must be YYYY-MM-DD HH:MM:SS
+        :qparam string end: Optionally specify a date to end at. Format must be YYYY-MM-DD HH:MM:SS
+        """
+        serializer = IncomingRequestSerializer(data=request.GET)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        serialized = serializer.validated_data
+
+        event_type = serialized.get("eventType")
+        errors_only = serialized.get("errorsOnly")
+        org_slug = serialized.get("organizationSlug")
+        start_time = serialized.get("start")
+        end_time = serialized.get("end")
+
+        organization = None
+        if org_slug:
+            try:
+                organization = OrganizationMapping.objects.get(slug=org_slug)
+            except OrganizationMapping.DoesNotExist:
+                return Response({"detail": "Invalid organization."}, status=400)
+
+        requests: list[BufferedRequest] = []
+        control_filter: SentryAppRequestFilterArgs = {}
+        region_filter: SentryAppRequestFilterArgs = {}
+        control_filter["errors_only"] = region_filter["errors_only"] = errors_only
+
+        def _filter_by_date(request: RpcSentryAppRequest, start: datetime, end: datetime) -> bool:
+            date_str = request.date
+            if not date_str:
+                return False
+            timestamp = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S.%f+00:00").replace(
+                microsecond=0, tzinfo=timezone.utc
+            )
+            return start <= timestamp <= end
+
+        def _filter_by_organization(
```

Comment:
is there any reason why these functions need to be defined inline in the endpoint get function? can they be moved out to their own scope?